helm-charts/chatqna/templates/deployment.yaml

```diff
@@ -33,6 +33,13 @@ spec:
         {{- toYaml .Values.podSecurityContext | nindent 8 }}
       containers:
         - name: {{ .Release.Name }}
+          {{- if .Values.externalLLM.enabled }}
```

Comment:
This big if/else leaves all other embedding/rerank/retriever related env variables non-set. That's the reason you got the the CI errors.

Maybe a better way is to override those settings under the following `env:` sections, by move this `envFrom` after the `env` section